plugins/UsersManager/tests/UI/UsersManager_AnonymousUser_spec.js

```diff
@@ -0,0 +1,139 @@
+/*!
+ * Matomo - free/libre analytics platform
+ *
+ * UsersManager screenshot tests.
+ *
+ * @link https://matomo.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ */
+
+describe('UsersManager_AnonymousUser', function () {
+  this.fixture = "Piwik\\Plugins\\UsersManager\\tests\\Fixtures\\AnonymousUser";
+
+  const url = "?module=UsersManager&action=index";
+
+  function getUserAccess(userName) {
+    return page.evaluate(
+      (userName) =>  {
+        return $(`#manageUsersTable #userLogin:contains(${userName}) + .access-cell li[role="option"][aria-selected="true"]`)
+          .text()
+          .trim()
+      },
+      userName
+    );
+  }
+
+  async function abortPasswordConfirmation() {
+    await page.$('.confirm-password-modal', { visible: true });
+    await (await page.jQuery('.confirm-password-modal.open .modal-close.modal-no:visible')).click();
+    await page.$('.confirm-password-modal', { hidden: true });
+  }
+
+  async function confirmPassword() {
+    await page.$('.confirm-password-modal', { visible: true });
+
+    await page.evaluate((superUserPassword) => {
+      $('.confirm-password-modal input[name=currentUserPassword]:visible')
+        .val(superUserPassword)
+        .change();
+    }, superUserPassword);
+
+    await page.waitForTimeout(100); // wait for input change to be registered
+    await (await page.jQuery('.confirm-password-modal .modal-close:not(.modal-no):visible')).click();
+    await page.$('.confirm-password-modal', { hidden: true });
+    await page.waitForNetworkIdle();
+  }
+
+  async function confirmRoleChange() {
+    await page.$('.change-user-role-confirm-modal.open', { visible: true });
+    await (await page.jQuery('.change-user-role-confirm-modal.open .modal-close:not(.modal-no):visible')).click();
+    await page.$('.change-user-role-confirm-modal.open', { hidden: true });
+    await page.waitForNetworkIdle();
+  }
+
+  it('should start with a known list of users', async function () {
+    await page.goto(url);
+    await page.waitForNetworkIdle();
+
+    expect(await getUserAccess('anonymous')).to.equal('No access');
+    expect(await getUserAccess('regularUser')).to.equal('No access');
+  });
+
+  describe('single user handling', function () {
+    async function setUserAccess(userName, accessString) {
+      await page.evaluate(
+        (userName, accessString) => {
+          $(`#manageUsersTable #userLogin:contains(${userName}) + .access-cell select`)
+            .val(accessString)
+            .change();
+        },
+        userName, accessString
+      )
+    }
+
+    it('should reset selected access if confirmation is aborted', async function () {
+      await setUserAccess('anonymous', 'string:view');
+      await abortPasswordConfirmation();
+
+      expect(await getUserAccess('anonymous')).to.equal('No access');
+    });
+
+    it('should show a password confirmation when giving access to anonymous user', async function () {
```

Comment:
Can we add two more tests for giving and revoking access to/for the regular user? Or is that covered elsewhere?

Reply:
Those cases should already be taken care of by the `UsersManager_spec` (including screenshots).

If we go the `OneUserPerRoleAndAnonymous` fixture route, we could take on the effort of splitting up the `UsersManager` specs into reasonably scoped chunks:

- `UsersManager_Invite`
- `UsersManager_ModifyBulk`
- `UsersManager_ModifySingleUser`
- `UsersManager_EditUser`
- `UsersManager_ManyUsers` (paging, paging search, bulk action for multiple pages)

And in combination maybe get rid of most of the screenshots. Checking the visual result of `clicked "Next" in pagination` is nice, but unless we are interested in potential rendering differences between first and other pages this could also just check the first user in the displayed list has changed.

Reply:
Ok, if the other behaviour is already covered elsewhere, this is fine for now. We could create an internal follow up ticket but that will likely be low priority so up to you, I'll merge this as is for now.